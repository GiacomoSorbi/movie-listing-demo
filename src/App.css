.app {
  text-align: center;
  min-height: 100vh;
}

.app-logo {
  width: 20%;
  height: 40vmin;
  background-image: url('./assets/film-roll.svg');
  background-repeat: repeat-y;
  background-position-x: center;
  background-size: contain;
  animation: app-logo-run infinite 60s linear;
  pointer-events: none;
}

.app-logo:last-child {
  animation-direction: reverse;
}

.app-header {
  background-color: lightyellow;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: calc(1s0px + 3vmin);
}

main {
  padding: 1vmin;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
}

@keyframes app-logo-run {
  from {
    background-position-y: -10000%;
  }
  to {
    background-position-y: 0%;
  }
}

@media only screen and (min-width: 768px) {
  .app-logo {
    background-size: cover;
    animation: app-logo-run infinite 40s linear;
  }
  main {
    flex-direction: row;
    align-items: stretch;
  }
}
