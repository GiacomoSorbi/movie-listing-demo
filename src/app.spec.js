import React from 'react'
import ReactDOM from 'react-dom'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './app'

describe('App should', () => {
  it('exist', () => {
    expect(App).toBeDefined()
  })

  it('render without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('have a consistent structure', () => {
    const wrapper = mount(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
