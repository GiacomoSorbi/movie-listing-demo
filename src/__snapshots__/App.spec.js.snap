// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should have a consistent structure 1`] = `
<App>
  <div
    className="app"
  >
    <AppHeader>
      <header
        className="app-header"
      >
        <div
          className="app-logo"
        />
        <h1>
          Currently airing movies
        </h1>
        <div
          className="app-logo"
        />
      </header>
    </AppHeader>
    <main>
      <Filters
        currentMoviesGenres={Array []}
        filters={Array []}
        genres={Array []}
        onChangeGenreFilters={[Function]}
        onChangeMinRating={[Function]}
      >
        <section
          className="movie-filters"
        >
          <div
            className="movie-filters-genres"
          />
          <input
            defaultValue="3"
            id="min-rating"
            max="10"
            min="0"
            onChange={[Function]}
            step="0.5"
            type="number"
          />
          <label
            htmlFor="min-rating"
          >
            Minimum Rating
          </label>
        </section>
      </Filters>
      <MovieList
        error={null}
        filters={Array []}
        genres={Array []}
        minRating={0}
        movies={Array []}
      >
        <section
          className="movie-list"
        />
      </MovieList>
    </main>
  </div>
</App>
`;
